### PR TITLE
feat(behavior_path_planner): consider object velocity direction

### PR DESCRIPTION
**Pull request by takayuki5168**
## Description

With the following PR, the vehicle object velocity direction will not be the same as the vehicle object orientation since the velocity center is not the rear wheel center but CoM.
https://github.com/autowarefoundation/autoware.universe/pull/4637

This PR considers the vehicle object velocity direction in a planning package.

## Tests performed

Unit test

## Effects on system behavior

Nothing

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by rej55 — APPROVED** [review]
LGTM

File: planning/behavior_path_planner/src/scene_module/dynamic_avoidance/dynamic_avoidance_module.cpp [hunk 1]
```diff
@@ -101,15 +101,20 @@ std::pair<double, double> projectObstacleVelocityToTrajectory(
...
+  const double obj_vel_norm = std::hypot(
+    object.kinematics.initial_twist_with_covariance.twist.linear.x,
+    object.kinematics.initial_twist_with_covariance.twist.linear.y);
 
   const size_t obj_idx = motion_utils::findNearestIndex(path_points, obj_pose.position);
 
-  const double obj_yaw = tf2::getYaw(obj_pose.orientation);
+  const double obj_vel_yaw = std::atan2(
+    object.kinematics.initial_twist_with_covariance.twist.linear.x,
+    object.kinematics.initial_twist_with_covariance.twist.linear.y);
```
Comment (takayuki5168) [inline planning/behavior_path_planner/src/scene_module/dynamic_avoidance/dynamic_avoidance_module.cpp:112]:
```suggestion
    object.kinematics.initial_twist_with_covariance.twist.linear.y,
    object.kinematics.initial_twist_with_covariance.twist.linear.x);
```

File: planning/behavior_path_planner/src/utils/utils.cpp [hunk 2]
```diff
@@ -2672,8 +2672,10 @@ PredictedObjects filterObjectsByVelocity(
   PredictedObjects filtered;
   filtered.header = objects.header;
   for (const auto & obj : objects.objects) {
-    const auto v = std::abs(obj.kinematics.initial_twist_with_covariance.twist.linear.x);
-    if (min_v < v && v < max_v) {
+    const auto v_norm = std::hypot(
+      obj.kinematics.initial_twist_with_covariance.twist.linear.x,
+      obj.kinematics.initial_twist_with_covariance.twist.linear.y);
+    if (min_v < v_norm && v_norm < max_v) {
```
Comment (kyoichi-sugahara) [inline planning/behavior_path_planner/src/utils/utils.cpp:2678]: I agree this change.
But I applied this change in this [PR](https://github.com/autowarefoundation/autoware.universe/pull/4658/files#diff-5e3a3ab9ddbf28c4acb9313e60b7d9f668773f2be8a3e12a1788ef476fce23a6R65-R68)
In this PR, this function is moved to path_safety_checker directory in utils. So could you revert this change?
```suggestion
    const auto v = std::abs(obj.kinematics.initial_twist_with_covariance.twist.linear.x);
    if (min_v < v && v < max_v) {
```

**Review by kyoichi-sugahara — APPROVED** [review]
LGTM
